error[E0271]: expected `TakeWhile<&mut IntoIter<u8>, {closure@issue-31173.rs:7:21}>` to be an iterator that yields `&_`, but it yields `u8`
  --> $DIR/issue-31173.rs:11:10
   |
LL |         .cloned()
   |          ^^^^^^ expected `&_`, found `u8`
   |
   = note: expected reference `&_`
                   found type `u8`
note: the method call chain might not have had the expected associated types
  --> $DIR/issue-31173.rs:3:20
   |
LL |   pub fn get_tok(it: &mut IntoIter<u8>) {
   |                      ^^^^^^^^^^^^^^^^^ `Iterator::Item` is `u8` here
...
LL |           .take_while(|&x| {
   |  __________-
LL | |             found_e = true;
LL | |             false
LL | |         })
   | |__________- `Iterator::Item` remains `u8` here
note: required by a bound in `cloned`
  --> $SRC_DIR/core/src/iter/traits/iterator.rs:LL:COL

error[E0599]: the method `collect` exists for struct `Cloned<TakeWhile<&mut IntoIter<u8>, {closure@issue-31173.rs:7:21}>>`, but its trait bounds were not satisfied
  --> $DIR/issue-31173.rs:12:10
   |
LL |       let temp: Vec<u8> = it
   |  _________________________-
LL | |         .take_while(|&x| {
LL | |             found_e = true;
LL | |             false
LL | |         })
LL | |         .cloned()
LL | |         .collect();
   | |         -^^^^^^^ method cannot be called due to unsatisfied trait bounds
   | |_________|
   |
   |
   = note: the following trait bounds were not satisfied:
           `<TakeWhile<&mut std::vec::IntoIter<u8>, {closure@$DIR/issue-31173.rs:7:21: 7:25}> as Iterator>::Item = &_`
           which is required by `Cloned<TakeWhile<&mut std::vec::IntoIter<u8>, {closure@$DIR/issue-31173.rs:7:21: 7:25}>>: Iterator`
           `Cloned<TakeWhile<&mut std::vec::IntoIter<u8>, {closure@$DIR/issue-31173.rs:7:21: 7:25}>>: Iterator`
           which is required by `&mut Cloned<TakeWhile<&mut std::vec::IntoIter<u8>, {closure@$DIR/issue-31173.rs:7:21: 7:25}>>: Iterator`

error: aborting due to 2 previous errors

Some errors have detailed explanations: E0271, E0599.
For more information about an error, try `rustc --explain E0271`.
